cornell-dti/carriage-web
# Finish ride modal

### Summary 






This pull request adds missing support for:

- Creating a daily, weekly, or custom repeating ride
- Editing a repeating ride (single instance or all repeating rides)

In addition, editing a ride on Rider Web is no longer broken.

### Test Plan 



Run the frontend, and try scheduling a repeating ride. After creation, only the parent repeating ride should be visible in the schedule.

When editing the repeating ride, you can either:

- Edit "this ride only" - the edits made will create a new non-repeating ride on the date specified, and the parent repeating ride start date will move to the next repeat occurrence
- Edit "all repeating rides" - the edits will be made to the parent repeating ride and will affect any future generated repeating rides

![image](https://user-images.githubusercontent.com/20343975/149634240-99ccc9b3-8ac3-48d0-b788-f67eb7246c60.png)
![image](https://user-images.githubusercontent.com/20343975/149634243-681b9407-a04d-46e4-bc74-fe1145b8342c.png)
![image](https://user-images.githubusercontent.com/20343975/149634247-2fd0729f-0085-4ab5-a39a-7b1e2dd4faaa.png)
![image](https://user-images.githubusercontent.com/20343975/149634298-89f67d7f-0a1e-4a9c-814c-ba227be7505d.png)


### Notes 



The `DaySelector` component I used has been created two other times in our codebase, but none of them are reusable without significant code refactoring. This should be fixed as soon as possible!

### Breaking Changes  



<!-- - Database schema change (anything that changes DynamoDB document structure)
<!-- - Other change that could cause problems (Detailed in notes)
